fix(dev): Correct errors in DI chapter code examples (#13840)
nextcloud/documentation: developer_manual/basics/dependency_injection.rst

Reviewer: ```suggestion
* Now **AuthorMapper** has all of its dependencies and can be constructed by the DI code. The object is returned.
```

Reviewer: ```suggestion
* **AuthorService** gets the **AuthorMapper**, is constructed, and returns the object
```

Reviewer: ```suggestion
* **AuthorService** is queried. This triggers the registered Callable::
```

core and app developers alike -- benefits from a more robust, secure, and future-proof
platform.

If you are unfamiliar with the DI design pattern, don't worry -- it's widely used in
modern frameworks, and you'll soon become comfortable with it. You can also watch the
following video introduction:

* `Google Clean Code Talks <https://www.youtube.com/watch?v=RlfLCWKxHJ0>`_

.. _dependency-injection:

Basic Pattern of Dependency Injection
-------------------------------------

The essence of dependency injection is: **don't instantiate dependencies directly inside
your classes or methods, but instead pass them in as parameters**. This allows swapping
out dependencies (such as with mocks in unit tests), makes dependencies explicit, and
centralizes object creation logic.

For example, consider the following pattern:

.. code-block:: php
  :emphasize-lines: 10, 14

  /** 
   * Without dependency injection:
   */

  use OCP\IDBConnection;

  class AuthorMapper {

    // Define a property to store the dependency
    private IDBConnection $db;

    public function __construct() {
      // The dependency is instantiated within the class

>       $this->db = new Db();

Reviewer 2: `\OC::$server` is deprecated @copilot.

    }
  }

> With dependency injection, you would instead pass the dependency into the constructor:

Reviewer: ```suggestion
With dependency injection, you would instead request the dependency as part of the constructor parameters:
```

.. code-block:: php
  :emphasize-lines: 10, 13, 15

  /**
   * Using dependency injection:
   */

  use OCP\IDBConnection;

  class AuthorMapper {

    // Define a property to store the dependency
    private IDBConnection $db;

    // The dependency is passed in from outside (typically by the container)
    public function __construct(IDBConnection $db) {
      // Assigned to the property
      $this->db = $db;
    }
  }

> Or, more succinctly, by using constructor property promotion (since PHP 7.4):

Reviewer: ```suggestion
Or, more succinctly, by using constructor property promotion (available in current PHP versions).
The following is exactly equivalent:
```
